contracts/src/v0.8/functions/tests/1_0_0/FunctionsRouter.t.sol

```diff
@@ -1,111 +1,354 @@
+
+}
+
+/// @notice #getAllowListId
+contract FunctionsRouter_GetAllowListId {
+
+}
+
+/// @notice #setAllowListId
+contract FunctionsRouter_SetAllowListId {
+
+}
+
+/// @notice #_getMaxConsumers
+contract FunctionsRouter__GetMaxConsumers {
+
+}
+
+/// @notice #sendRequest
+contract FunctionsRouter_SendRequest {
+
+}
+
+/// @notice #sendRequestToProposed
+contract FunctionsRouter_SendRequestToProposed {
+
+}
+
+/// @notice #_sendRequest
+contract FunctionsRouter__SendRequest {
+
+}
+
+/// @notice #fulfill
+contract FunctionsRouter_Fulfill {
+
+}
+
+/// @notice #_callback
+contract FunctionsRouter__Callback {
+
+}
+
+/// @notice #getContractById
+contract FunctionsRouter_GetContractById {
+
+}
 
-  function getRouterConfig() public view returns (FunctionsRouter.Config memory) {
-    uint32[] memory maxCallbackGasLimits = new uint32[](1);
-    maxCallbackGasLimits[0] = type(uint32).max;
-
-    return
-      FunctionsRouter.Config({
-        maxConsumersPerSubscription: s_maxConsumersPerSubscription,
-        adminFee: s_adminFee,
-        handleOracleFulfillmentSelector: s_handleOracleFulfillmentSelector,
-        maxCallbackGasLimits: maxCallbackGasLimits,
-        gasForCallExactCheck: 5000
-      });
+/// @notice #getProposedContractById
+contract FunctionsRouter_GetProposedContractById {
+
+}
+
+/// @notice #getProposedContractSet
+contract FunctionsRouter_GetProposedContractSet {
+
+}
+
+/// @notice #proposeContractsUpdate
+contract FunctionsRouter_ProposeContractsUpdate {
+
+}
+
+/// @notice #updateContracts
+contract FunctionsRouter_UpdateContracts {
+
+}
+
+/// @notice #_whenNotPaused
+contract FunctionsRouter__WhenNotPaused {
+
+}
+
+/// @notice #_onlyRouterOwner
+contract FunctionsRouter__OnlyRouterOwner {
+
+}
+
+/// @notice #_onlySenderThatAcceptedToS
+contract FunctionsRouter__OnlySenderThatAcceptedToS {
+
+}
+
+/// @notice #pause
+contract FunctionsRouter_Pause is FunctionsRouterSetup {
+  function setUp() public virtual override {
+    FunctionsRouterSetup.setUp();
   }
 
-  function getCoordinatorConfig() public pure returns (FunctionsBilling.Config memory) {
-    return
-      FunctionsBilling.Config({
-        maxCallbackGasLimit: 5,
-        feedStalenessSeconds: 5,
-        gasOverheadAfterCallback: 5,
-        gasOverheadBeforeCallback: 5,
-        requestTimeoutSeconds: 1,
-        donFee: 5,
-        maxSupportedRequestDataVersion: 5,
-        fulfillmentGasPriceOverEstimationBP: 5,
-        fallbackNativePerUnitLink: 2874
-      });
+  event Paused(address account);
+
+  function test_Pause_RevertIfNotOwner() public {
+    // Send as stranger
+    vm.stopPrank();
+    vm.startPrank(STRANGER_ADDRESS);
+
+    vm.expectRevert("Only callable by owner");
+    s_functionsRouter.pause();
   }
 
-  function getTermsOfServiceConfig() public pure returns (TermsOfServiceAllowList.Config memory) {
-    return TermsOfServiceAllowList.Config({enabled: false, signerPublicKey: address(132)});
+  function test_Pause_Success() public {
```

Comment:
Maybe check that sendRequest reverts?

Reply:
I'm had planned to leave that to the concern of each method, but you are right that it probably doesn't hurt to have it.